File: leetcode/Accepted/Easy/sum-of-all-odd-length-subarrays.cpp
class Solution {
 public:
  int sumOddLengthSubarrays(vector<int>& arr) {
    int ans = 0, n = (int)arr.size();
    for (int i = 0; i < n; i++) {
      int occur = ceil(((i + 1) * (n - i)) / 2.0);
      ans += (arr[i] * occur);
    }
    return ans;
  }
};

File: leetcode/AC/Easy/sum-of-all-odd-length-subarrays.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        ans = 0
        n = len(arr)
        for i in range(n):
            occur = ceil(((i+1)*(n-i))/2.0)
            ans += (occur * arr[i])

        return ans
